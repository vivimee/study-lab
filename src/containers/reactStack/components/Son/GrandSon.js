import React, { Component } from "react";
import PropTypes from "prop-types";
import { AgeConsumer } from "../..";

class GrandSon extends Component {
  static contextTypes = {
    name: PropTypes.string
  };
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return (
      <div>
        <p>this is GrandSon. GrandParent name: {this.context.name}.</p>
        <AgeConsumer>
          {({ age, gender }) => (
            <p>
              {age}-{gender}
            </p>
          )}
        </AgeConsumer>
      </div>
    );
  }
}

export default GrandSon;
