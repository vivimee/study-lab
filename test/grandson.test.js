import React from 'react';
import renderer from 'react-test-renderer';
import GrandSon from '../src/containers/reactStack/components/Son/GrandSon';

test('GrandSon should has name and age', () => {
  const component = renderer.create(<GrandSon />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
})
