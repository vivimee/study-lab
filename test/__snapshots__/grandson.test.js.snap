// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrandSon should has name and age 1`] = `
<div>
  <p>
    this is GrandSon. GrandParent name: 
    .
  </p>
  <p>
    20
    -
  </p>
</div>
`;
